native/src/binaryRead.h

```diff
@@ -308,6 +309,23 @@ struct RouteDataObject {
 		return false;
 	}
 
+	void removePointType(int ind, int type) {
+		if (ind < pointTypes.size()) {
+			std::vector<uint32_t> typesArr = pointTypes[ind];
+			for (int i = 0; i < typesArr.size(); i++) {
+				if (typesArr[i] == type) {
+					std::vector<uint32_t> result;
+					std::copy(std::begin(typesArr), std::end(typesArr), std::begin(result) + i);
```

Comment:
Call to copy does not correspond to java's arraycopy(Object src, int srcPos, Object dest, int destPos, int length).
std::copy(begin, end, position in target vector). In other words, destPos in java is 0, but this code will insert starting from i-th position which is incorrect.